This is what I get when tried to build the app with the instructions from
CONTRIBUTING.md

npm ERR! companion@0.1.0 cordova-setup: `gulp cordova:setup`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the companion@0.1.0 cordova-setup script.
npm ERR! This is most likely a problem with the companion package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     gulp cordova:setup
npm ERR! You can get their info via:
npm ERR!     npm owner ls companion
npm ERR! There is likely additional logging output above.
npm ERR! System Darwin 12.5.0
npm ERR! command "node" "/usr/local/bin/npm" "run-script" "cordova-setup"
npm ERR! cwd
/Users/marciogalli/Desktop/fisl/2014/2014-experiments/formal/companion
npm ERR! node -v v0.10.26
npm ERR! npm -v 1.4.3
npm ERR! code ELIFECYCLE
npm ERR!
npm ERR! Additional logging details can be found in:
npm ERR!
/Users/marciogalli/Desktop/fisl/2014/2014-experiments/formal/companion/npm-debug.log
npm ERR! not ok code 0

On Tue, Apr 8, 2014 at 4:44 AM, Fabricio C Zuardi
notifications@github.comwrote:

> only cordova create task is functional ATM, but if you feel that it is on
> 
> ## the right path feel free to merge
> 
> You can merge this Pull Request by running
> 
>   git pull https://github.com/TelaSocial/companion cordova-buildsystem
> 
> Or view, comment on, or merge it at:
> 
>   https://github.com/TelaSocial/companion/pull/3
> Commit Summary
> - start of a build system that will setup cordova files under
>   /dist/cordova/
> 
> File Changes
> - _A_ .gitignorehttps://github.com/TelaSocial/companion/pull/3/files#diff-0(3)
> - _A_ CONTRIBUTING.mdhttps://github.com/TelaSocial/companion/pull/3/files#diff-1(23)
> - _A_ gulpfile.jshttps://github.com/TelaSocial/companion/pull/3/files#diff-2(72)
> - _A_ package.jsonhttps://github.com/TelaSocial/companion/pull/3/files#diff-3(44)
> 
> Patch Links:
> - https://github.com/TelaSocial/companion/pull/3.patch
> - https://github.com/TelaSocial/companion/pull/3.diff
> 
> ## 
> 
> Reply to this email directly or view it on GitHubhttps://github.com/TelaSocial/companion/pull/3
> .

--- a/gulpfile.js
+++ b/gulpfile.js
@@ -2,10 +2,13 @@
 
 //libraries
 var pkg = require('./package.json'),
+    path = require('path'),
+    mkdirp = require('mkdirp'),
     gulp = require('gulp'),
+    gutil = require('gulp-util'),
     cordova = require('cordova'),
     CordovaError = require('cordova/src/CordovaError'),
-    gutil = require('gulp-util');
+    Q = require('q');
 
 //paths
 var paths = {
@@ -16,40 +19,61 @@ var paths = {
             data: 'src/data/*.json'
         },
         build: {
+            root: 'dist',
             www: 'dist/www/',
             cordova: 'dist/cordova/',
         }
     };
 
 //globals
-var magenta = gutil.colors.magenta,
-    red = gutil.colors.red,
+var red = gutil.colors.red,
+    // magenta = gutil.colors.magenta,
     // cyan = gutil.colors.cyan,
-    // blue = gutil.colors.blue,
-    green = gutil.colors.green;
+    // green = gutil.colors.green,
+    blue = gutil.colors.blue;
 
 //----------------------------------------------------------------------------
 // tasks
 //----------------------------------------------------------------------------
 
+gulp.task('createBuildDir', function(cb){
+    mkdirp(paths.build.root, function(err) {
+        if(err){ throw err; }
+        cb(null);
+    });
+});
+
+var cordovaLog = function(msg){
+    console.log('[' + blue('cordova')+ '] ', msg);
+};
+
 //creates a cordova folder for the app
-gulp.task('cordova:create', function(){
+gulp.task('cordova:create', ['createBuildDir'], function(cb){
     var name = pkg.name,
         dir = paths.build.cordova,
         id = pkg.cordovaConfig.id,
-        cfg = pkg.cordovaConfig.extra;
+        cfg = pkg.cordovaConfig.extra,
+        promise;
 
-    gutil.log(
-        'Setting up', magenta(name),'(' + green(id) + ') at', magenta(dir) + '...'
-    );
-    cordova.create(dir, id, name, cfg);
+    cordova.on('log', cordovaLog);
+    cordova.on('verbose', cordovaLog);
+    promise = new Q(cordova.create(dir, id, name, cfg));
+    promise.then(function(){
+        gutil.log('cordova create finished.');
+        cb(null);
+    });
+});
+
+gulp.task('cordova:addAndroid', ['cordova:create'], function(){
+    process.env.PWD = path.join(process.env.PWD, paths.build.cordova);
+    console.log('process.env.PWD', process.env.PWD);
+    cordova.platform('add', 'android');
 });
 
 //run all the initial cordova-related tasks
 gulp.task('cordova:setup',
     [
-        'cordova:create'
-// - cordova platform add android
+        'cordova:addAndroid'
 // - cordova plugin add https://github.com/EddyVerbruggen/Calendar-PhoneGap-Plugin.git
 // - cordova plugin add https://github.com/Red-Folder/bgs-core
 // - cordova plugin add https://github.com/Red-Folder/bgs-sample

--- a/package.json
+++ b/package.json
@@ -28,7 +28,9 @@
   "devDependencies": {
     "cordova": "^3.4.0-0.1.3",
     "gulp": "^3.6.0",
-    "gulp-util": "^2.2.14"
+    "gulp-util": "^2.2.14",
+    "mkdirp": "^0.3.5",
+    "q": "^1.0.1"
   },
   "jshintConfig": {
     "globals": {

